metals/src/main/scala/scala/meta/internal/metals/BuildServerConnection.scala

```diff
@@ -183,17 +183,22 @@ class BuildServerConnection private (
   def jvmRunEnvironment(
       params: JvmRunEnvironmentParams
   ): Future[JvmRunEnvironmentResult] = {
+    def empty = new JvmRunEnvironmentResult(Collections.emptyList)
     connection.flatMap { conn =>
-      if (
-        conn.capabilities
-          .getJvmRunEnvironmentProvider() && conn.displayName != "scala-cli"
-      ) {
-        register(server => server.jvmRunEnvironment(params)).asScala
+      if (conn.capabilities.getJvmRunEnvironmentProvider()) {
+        register(
+          server => server.jvmRunEnvironment(params),
+          onFail = Some(
+            (
+              empty,
+              s"${name} should support `buildTarget/jvmRunEnvironment`, but it fails.",
+            )
+          ),
+        ).asScala
       } else {
-        scribe.warn(
+        scribe.debug(
```

Comment:
Why such change? It shouldn't be logged that often since most common build server support that endpoint. Isn't warn a proper severity for something which is a bit unexpected?

Reply:
I thought that it might be confusing for people, but maybe it's fine.

Reply:
Reverted!

Reply:
sbt, bloop, scala-cli and mill supports it, I think that else branch is highly unlikely to be executed